src/main/java/org/opensearch/sdk/handlers/LocalNodeResponseHandler.java

```diff
@@ -12,29 +12,48 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.opensearch.cluster.LocalNodeResponse;
+import org.opensearch.cluster.node.DiscoveryNode;
 import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.extensions.ExtensionsOrchestrator;
 import org.opensearch.sdk.ExtensionsRunner;
 import org.opensearch.threadpool.ThreadPool;
 import org.opensearch.transport.TransportException;
 import org.opensearch.transport.TransportResponseHandler;
 import org.opensearch.transport.TransportService;
 
 import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
 
 /**
  * This class handles the response from OpenSearch to a {@link ExtensionsRunner#sendLocalNodeRequest(TransportService)} call.
  */
 public class LocalNodeResponseHandler implements TransportResponseHandler<LocalNodeResponse> {
     private static final Logger logger = LogManager.getLogger(LocalNodeResponseHandler.class);
+    private final CountDownLatch inProgressLatch;
+    private DiscoveryNode localNode;
+
+    /**
+     * Instantiates a new LocalNodeResponseHandler with a count down latch
+     */
+    public LocalNodeResponseHandler() {
+        this.inProgressLatch = new CountDownLatch(1);
+        this.localNode = null;
```

Comment:
Since we are setting it to `null`? Do we need it in constructor? We can just have it as a class variable

Reply:
We'll need it in the constructor because we'll update this upon receiving a response

Reply:
From a style perspective I'd really like to avoid setting to null in the constructor.  It's basically a no-op.  You can just add an `= null` on line 34 and leave it off here.